### DISCUSSION TITLE
Routing based on HTTP methods

### DISCUSSION THREAD

**Discussion by stasmir** [projectcontour/contour#4999] [General]
Is it possible to create routes based on a HTTP method, like to route all GET to a service A and POST, PUT, DELETE to a service B?

**Comment by sunjayBhatia** [comment]
Yes, you can use a `:method` header match to set up routing rules based on HTTP method

**Reply by stasmir** [reply]
How to specify several method in this case?

- header:
      method: POST, PUT, DELETE  

And is it available in 1.18.2 version?

**Reply by stasmir** [reply]
Found an example https://github.com/projectcontour/contour/issues/2679#issuecomment-844300112

**Reply by skriss** [reply]
If you want to match multiple methods, you'll need a separate route for each. And yes, this should be available in 1.18.2 since it's using the existing header matching capability (though we'd encourage you to upgrade to a newer, supported version when you can!)
